pkg/theia/commands/clickhouse_status.go

```diff
@@ -0,0 +1,308 @@
+				fmt.Println(arr)
+			}
+		}
+	}
+	if options.insertRate {
+		data, err := getWritingRateFromClickHouse(connect)
+		if err != nil {
+			return err
+		}
+		if options.printTable {
+			printTable(data)
+		} else {
+			for _, arr := range data {
+				fmt.Println(arr)
+			}
+		}
+	}
+	if options.stackTraces {
+		data, err := getStackTracesFromClickHouse(connect)
+		if err != nil {
+			return err
+		}
+		if options.printTable {
+			printTable(data)
+		} else {
+			for _, arr := range data {
+				fmt.Println(arr)
+			}
+		}
+	}
+	return nil
+}
+
+func getDiskInfoFromClickHouse(connect *sql.DB) ([][]string, error) {
+	result, err := connect.Query(diskQuery)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get disk information: %v", err)
+	}
+	defer result.Close()
+	columnName, err := result.Columns()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get the head of disk information: %v", err)
+	}
+	var data [][]string
+	data = append(data, columnName)
+	for result.Next() {
+		var res diskInfo
+		result.Scan(&res.shard, &res.name, &res.path, &res.freeSpace, &res.totalSpace, &res.usedPercentage)
+		data = append(data, []string{res.shard, res.name, res.path, res.freeSpace, res.totalSpace, res.usedPercentage})
+	}
+	if len(data) <= 1 {
+		return nil, fmt.Errorf("no data is returned by database")
+	}
+	return data, nil
+}
+
+func getTableInfoBasicFromClickHouse(connect *sql.DB) ([][]string, error) {
+	result, err := connect.Query(tableInfoBasicQuery)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get table information: %v", err)
+	}
+	defer result.Close()
+	columnName, err := result.Columns()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get the head of table information: %v", err)
+	}
+	var data [][]string
+	data = append(data, columnName)
+	for result.Next() {
+		res := tableInfoBasic{}
+		result.Scan(&res.shard, &res.database, &res.tableName, &res.totalRows, &res.totalBytes, &res.totalCols)
+		data = append(data, []string{res.shard, res.database, res.tableName, res.totalRows, res.totalBytes, res.totalCols})
+	}
+	if len(data) <= 1 {
+		return nil, fmt.Errorf("no data is returned by database")
+	}
+	return data, nil
+}
+
+func getWritingRateFromClickHouse(connect *sql.DB) ([][]string, error) {
+	result, err := connect.Query(writePerSecQuery)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get insertion rate: %v", err)
+	}
+	defer result.Close()
+	columnName, err := result.Columns()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get the head of insertion rate: %v", err)
+	}
+	var data [][]string
+	data = append(data, columnName)
+	for result.Next() {
+		res := writeRowsPerSec{}
+		result.Scan(&res.shard, &res.rowsPerSec, &res.bytesPerSec)
+		data = append(data, []string{res.shard, res.rowsPerSec, res.bytesPerSec})
+	}
+	if len(data) <= 1 {
+		return nil, fmt.Errorf("no data is returned by database")
+	}
+	return data, nil
+}
+
+func getStackTracesFromClickHouse(connect *sql.DB) ([][]string, error) {
```

Comment:
If ClickHouse is busy with something and you don’t know what’s happening, you can check the stacktraces of all the thread which are working. I talked to Salvatore before, he think it might be helpful to add this option.